### PR TITLE
TikiWiki reader: improve list parsing

### LINKED ISSUE #4722: Tikiwiki reader should parse lists that have no space after list marker
when evaluating tikwiki, an unordered list is not evaluated.

example:

*list
**will not
**evaluate

### PR DESCRIPTION
**Pull request by mb21**
- remove trailing Space from list items
- also parse lists that have no space after list marker (fixes #4722)

@rlpowell what do you think?

### CODE REVIEW

**Comment by rlpowell** [comment]
Assuming that test/command/4722.md is run as an automated test, this
looks excellent to me, thank you.

File: src/Text/Pandoc/Readers/TikiWiki.hs [hunk 1]
```diff
@@ -344,9 +344,13 @@ listItemLine nest = lineContent >>= parseContent
       return $ filterSpaces content ++ "\n" ++ Data.Maybe.fromMaybe "" continuation
     filterSpaces = reverse . dropWhile (== ' ') . reverse
     listContinuation = string (replicate nest '+') >> lineContent
+    removeSpace x = if B.toList x == [Space]
+                    then []
+                    else [x]
     parseContent x = do
       parsed <- parseFromString (many1 inline) x
-      return $ mconcat parsed
+      let ins = init parsed ++ removeSpace (last parsed)
```
Comment (jgm) [inline src/Text/Pandoc/Readers/TikiWiki.hs]: Generally I think we should avoid using partial functions like `init` and `last` whenever possible.  These crash with a runtime error when the list is empty.  Even if you know the list won't be empty when you write this code, sometimes a later change will cause the assumption to be violated, so it's good to code defensively.

**Comment by mb21** [comment]
@jgm, right... removed partial functions...

File: src/Text/Pandoc/Readers/TikiWiki.hs [hunk 2]
```diff
@@ -344,9 +344,14 @@ listItemLine nest = lineContent >>= parseContent
       return $ filterSpaces content ++ "\n" ++ Data.Maybe.fromMaybe "" continuation
     filterSpaces = reverse . dropWhile (== ' ') . reverse
     listContinuation = string (replicate nest '+') >> lineContent
+    removeTrailingSpace []     = []
```
Comment (jgm) [inline src/Text/Pandoc/Readers/TikiWiki.hs]: Could `dropWhileEnd` from Data.List be used instead?

File: src/Text/Pandoc/Readers/TikiWiki.hs [hunk 3]
```diff
@@ -344,9 +344,14 @@ listItemLine nest = lineContent >>= parseContent
...
+    removeTrailingSpace (x:[]) = if x == B.space
```
Comment (jgm) [inline src/Text/Pandoc/Readers/TikiWiki.hs]: Assuming this isn't replaced with `dropWhileEnd`, this seems unnecessarily wordy.  Why not
```
removeTrailingSpace [B.space] = ...
```

**Comment by mb21** [comment]
Yes! `dropWhileEnd` was what I was looking for... thanks.

**Pull request merged**